src/python/ensembl/io/genomio/gff3/process_gff3.py

```diff
@@ -400,99 +424,67 @@ def normalize_gene(
-
-                    # Replace qualifiers
-                    feat.qualifiers = {
-                        "ID": feat.id,
-                        "Parent": transcript.id,
-                        "phase": feat.qualifiers["phase"],
-                        "source": feat.qualifiers["source"],
-                    }
-                else:
-                    if feat.type in ignored_transcript_types:
-                        exons_to_delete.append(tcount)
-                        continue
-
-                    fail_types["sub_transcript=" + feat.type] = 1
-                    message = (
-                        f"Unrecognized exon type for {feat.type}: {feat.id}"
-                        f" (for transcript {transcript.id} of type {transcript.type})"
-                    )
-                    print(message)
-                    if self.skip_unrecognized:
-                        exons_to_delete.append(tcount)
-                        continue
-
-            if exons_to_delete:
-                for elt in sorted(exons_to_delete, reverse=True):
-                    transcript.sub_features.pop(elt)
+            transcript = self._normalize_transcript_subfeatures(gene, transcript, fail_types)
 
         if transcripts_to_delete:
             for elt in sorted(transcripts_to_delete, reverse=True):
                 gene.sub_features.pop(elt)
 
-        # PSEUDOGENE CDS IDs
-        if gene.type == "pseudogene" and self.allow_pseudogene_with_CDS:
-            self.normalize_pseudogene_cds(gene)
-
-        # Finally, store gene functional annotation
-        self.add_funcann_feature(functional_annotation, gene, "gene")
-
-        # replace qualifiers
-        old_gene_qualifiers = gene.qualifiers
-        gene.qualifiers = {"ID": gene.id, "source": old_gene_qualifiers["source"]}
-
         return gene
 
-    def clean_functional_annotations(self, functional_annotation: List) -> List:
-        """Returns the functional annotations list without putative product descriptions."""
-        for feat in functional_annotation:
-            if "description" in feat and not self.check_product(feat["description"]):
-                del feat["description"]
-        return functional_annotation
-
-    def check_product(self, product: str) -> bool:
-        """Check a product string.
-
-        Args:
-            product: a product name.
+    def _normalize_transcript_subfeatures(
+        self, gene: SeqFeature, transcript: SeqFeature, fail_types
+    ) -> SeqFeature:
+        """Returns a transcript with normalized sub-features."""
+        ignored_transcript_types = self.ignored_transcript_types
+        cds_found = False
+        exons_to_delete = []
+        for tcount, feat in enumerate(transcript.sub_features):
+            if feat.type == "exon":
+                # Replace qualifiers
+                old_exon_qualifiers = feat.qualifiers
+                feat.qualifiers = {
+                    "Parent": transcript.id,
+                }
```

Comment:
```suggestion
                feat.qualifiers = {"Parent": transcript.id}
```
Unless we plan to expand it in the future.